src/node.c

```diff
@@ -30,9 +30,6 @@ static bool S_can_contain(cmark_node *node, cmark_node *child) {
   if (node == NULL || child == NULL) {
     return false;
   }
-  if (NODE_MEM(node) != NODE_MEM(child)) {
```

Comment:
To be honest after reviewing these new additions, I'm very perplex as to:

* Why the code was complexified to pass this `cmark_mem` structure all around
* Why different nodes in the same tree should have the same "allocator", which is only a thin wrapper calling abort if the allocation failed anyway.

Note that AFAIU on most linux boxes, checking the return of memory allocation syscalls doesn't make much sense anyway.